awx/main/models/credential/__init__.py

```diff
@@ -1093,6 +1097,38 @@ def create(self):
 )
 
 
+ManagedCredentialType(
```

Comment:
We are introducing a new credential type but only making it useable when attached to an instance group. I would imagine this could be useful to customers outside of this context, in their job runs. Should the logic that lives in `kubectl_or_oc_vars` be moved to our general credential type injection so this cred can be re-used? @ryanpetrello @AlanCoding opinions?

Reply:
If it cannot be reused, then we should add this to the credential validation such that the "kubernetes" kind cannot be attached to inventory syncs or job runs. There is not a great standard pattern for this.

I was actually very excited about implementing either the [openshift](https://docs.ansible.com/ansible/latest/plugins/inventory/openshift.html) or [k8s](https://docs.ansible.com/ansible/latest/plugins/inventory/k8s.html) inventory plugin, and of course this would use the official credential type. I have some history on that. I was really trying to avoid writing the kubeconfig file, because the plugin itself should be able to process the token. Link https://github.com/openshift/openshift-restclient-python/issues/249

I do need to go back through see how this lays down the credential information and compare to what we might do with module / inventory plugin injection.

Reply:
@chrismeyersfsu how would the user go about attaching this credential to jobs (job templates?) in the extended scenarios you're speaking of?

Reply:
Would fit into our existing cloud credential(s).